src/scope.litcoffee

```diff
@@ -78,11 +77,17 @@ Generate a temporary variable name at the given index.
         else
           "#{name}#{index or ''}"
 
-Gets the type of a variable.
+Gets a variable and its associated data from this scope (not ancestors),
+or `undefined` if it doesn't exist.
+
+      get: (name) ->
+        @variables[@positions[name]] if Object::hasOwnProperty.call @positions, name
+
+Gets the type of a variable declared in this scope,
+or `undefined` if it doesn't exist.
 
       type: (name) ->
-        return v.type for v in @variables when v.name is name
-        null
+        @get(name)?.type
```

Comment:
I know this isn’t code that you added, but what’s a variable’s “type”? Can we explain that in this comment?